src/Fixer/Import/FullyQualifiedStrictTypesFixer.php

```diff
@@ -447,17 +447,40 @@ private function setupUsesFromDiscoveredSymbols(array &$uses, string $namespaceN
                 $discoveredFqcnByShortNameLower[strtolower($useShortName)] = $useLongName;
             }
 
-            uasort($discoveredSymbols, static fn ($a, $b) => substr_count($a, '\\') <=> substr_count($b, '\\'));
+            $useByShortNameLower = [];
+            foreach ($uses as $useShortName) {
+                $useByShortNameLower[strtolower($useShortName)] = true;
+            }
+
+            uasort($discoveredSymbols, static function ($a, $b) {
+                $res = str_starts_with($a, '\\') <=> str_starts_with($b, '\\');
+                if (0 !== $res) {
+                    return $res;
+                }
+
+                return substr_count($a, '\\') <=> substr_count($b, '\\');
```

Comment:
Why do we sort by `\` occurrences? Maybe it would be good to add a comment before `uasort` to describe what it solves.

Reply:
This is tough part of the fixer. I feel like we should rewrite this part later. Currently, we try to prevent conflicts by iterating with shortest relative name to prevent conflicts defined by the current tests.